faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -121,6 +121,12 @@ class ClientSpec extends FlatSpec with Matchers with BeforeAndAfterAll {
     await(client.query("qux")) should equal (StringV("qux"))
   }
 
+  it should "fail with timeout error" in {
+    val timeout = Duration.Zero
+    val thrown = the [UnavailableException] thrownBy await(client.query("echo", timeout))
+    thrown.message should include ("time out: Read timed out.")
+  }
```

Comment:
can we add some tests for the headers as well?

Reply:
Yes, we discussed that! We need to set up a unit test with mocking features in order to do so (and for testing another things beyond the scope of this ticket which are missing), but since there's none for the time being, it will be handled in the following ticket: https://faunadb.atlassian.net/browse/DRV-146